spec/unit/puppet/provider/registry_value_spec.rb

```diff
@@ -1,12 +1,45 @@
 #! /usr/bin/env ruby
 
 require 'spec_helper'
+require 'puppet/type/registry_key'
 require 'puppet/type/registry_value'
 
 describe Puppet::Type.type(:registry_value).provider(:registry), :if => Puppet.features.microsoft_windows? do
   let (:catalog) do Puppet::Resource::Catalog.new end
   let (:type) { Puppet::Type.type(:registry_value) }
 
+  puppet_key = "SOFTWARE\\Puppet Labs"
+  subkey_name ="PuppetRegProviderTest"
+
+  before(:all) do
+    Win32::Registry::HKEY_LOCAL_MACHINE.create("#{puppet_key}\\#{subkey_name}",
+      Win32::Registry::KEY_ALL_ACCESS)
+  end
+
+  before(:each) do
+    # problematic Ruby codepath triggers a conversion of UTF-16LE to
+    # a local codepage which can totally break when that codepage has no
+    # conversion from the given UTF-16LE characters to local codepage
+    # a prime example is that IBM437 has no conversion from a Unicode en-dash
+    Win32::Registry.expects(:export_string).never
+
+    # also, expect that we're not using Rubys each_key / each_value which exhibit bad behavior
+    Win32::Registry.expects(:each_key).never
+    Win32::Registry.expects(:each_value).never
+  end
+
+  after(:all) do
+    # Ruby 2.1.5 has bugs with deleting registry keys due to using ANSI
+    # character APIs, but passing wide strings to them (facepalm)
+    # https://github.com/ruby/ruby/blob/v2_1_5/ext/win32/lib/win32/registry.rb#L323-L329
+    # therefore, use our own code instead of hklm.delete_value
+
+    # TODO: don't like registry_value tests depending on registry_key type
```

Comment:
For 9db94e2, this is the only easy way of doing this. In a subsequent commit this could be rewritten to use the methods on ProviderBase instead, but not sure that's worth it.